File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> aa(26,0);
        

        if(s.length()!= t.length()){
            return false;
        }

        for( char ch : s){
            aa[ch - 'a']++;
        }

         for( char ch : t){
            aa[ch - 'a']--;
        }

         for( int cnt : aa){
            if(cnt != 0){
                return false;
            }
        }

        return true;

    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    # 2. Valid Anagram (LeetCode 242)
    def isAnagram(self, s: str, t: str) -> bool:
        # If lengths differ, they cannot be anagrams
        if len(s) != len(t):
            return False
        
        # Count frequencies of characters for both strings
        count_s, count_t = {}, {}
        for i in range(len(s)):
            count_s[s[i]] = count_s.get(s[i], 0) + 1
            count_t[t[i]] = count_t.get(t[i], 0) + 1
            
        # Compare frequency dictionaries
        return count_s == count_t